core/utils/src/main/kotlin/edu/stanford/spezi/core/utils/foundation/RepositoryAnchor.kt

```diff
@@ -0,0 +1,3 @@
+package edu.stanford.spezi.core.utils.foundation
+
+interface RepositoryAnchor
```

Comment:
The Anchor constraint is definitely used. However, it is not used in the interface of KnowledgeSource, but rather to ensure a 1-to-many mapping between anchor (i.e. application area) of a knowledgesource (e.g. you can only use this key for account configuration and not for onboarding stuff). The values should not be linked to that anchor, since for example, it may make sense to store String values in both account configuration and also onboarding stuff (to continue on with the example before).

Reply:
as discussed / written below the other message